src/ioctls/vm.rs

```diff
@@ -1237,6 +1237,201 @@ impl VmFd {
     pub fn check_extension(&self, c: Cap) -> bool {
         self.check_extension_int(c) > 0
     }
+
+    /// Issues platform-specific memory encryption commands to manage encrypted VMs if
+    /// the platform supports creating those encrypted VMs.
+    ///
+    /// Currently, this ioctl is used for issuing Secure Encrypted Virtualization
+    /// (SEV) commands on AMD Processors.
+    ///
+    /// See the documentation for `KVM_MEMORY_ENCRYPT_OP`.
+    ///
+    /// # Safety
+    ///
+    /// This function is unsafe because there is no guarantee `T` is valid in this context, how
+    /// much data kernel will read from memory and where it will write data on error.
+    ///
+    /// # Arguments
+    ///
+    /// * `op` - an opaque platform specific structure.
+    ///
+    /// # Example
+    ///
+    /// ```rust,no_run
+    /// # extern crate kvm_ioctls;
+    /// # extern crate kvm_bindings;
+    /// use kvm_bindings::bindings::kvm_sev_cmd;
+    /// # use kvm_ioctls::Kvm;
+    ///
+    /// let kvm = Kvm::new().unwrap();
+    /// let vm = kvm.create_vm().unwrap();
+    ///
+    /// // Initialize the SEV platform context.
+    /// let mut init: kvm_sev_cmd = Default::default();
+    /// unsafe { vm.encrypt_op(&mut init).unwrap() };
+    /// ```
+    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+    pub unsafe fn encrypt_op<T>(&self, op: &mut T) -> Result<()> {
```

Comment:
I added `encrypt_opt_sev` and `is_sev_enabled`, however I kept the unsafe, generic `encrypt_op` public, because AFAIK, contrary to the kernel documentation this particular ioctl is used for functionality not limited to SEV.
@haraldh maybe you could add some context here?

Reply:
`is_sev_enabled()` would need much more checks than for the `KVM_MEMORY_ENCRYPT_OP` to succeed.

Also as later mentioned in this PR, other technologies can and will be using `KVM_MEMORY_ENCRYPT_OP`.

Reply:
See https://github.com/intel/tdx/blob/kvm/Documentation/virt/kvm/intel-tdx.rst

Reply:
thanks